docs/sources/enterprise/vault.md

```diff
@@ -0,0 +1,136 @@
+set up with a limited lifetime.
+
+If you're using short-lived leases, you can also configure how often Grafana should renew the lease and for
+how long. We recommend keeping the defaults here unless you run into problems.
+
+```ini
+[keystore.vault]
+# Location of the Vault server
+;url =
+# Vault's namespace if using Vault with multi-tenancy
+;namespace =
+# Method for authenticating towards Vault. Vault is inactive if this option is not set
+# Possible values: token
+;auth_method =
+# Secret token to connect to Vault when auth_method is token
+;token =
+# Time between checking if there are any secrets which needs to be renewed.
+;lease_renewal_interval = 5m
+# Time until expiration for tokens which are renewed. Should have a value higher than lease_renewal_interval
+;lease_renewal_expires_within = 15m
+# New duration for renewed tokens. Vault may be configured to ignore this value and impose a stricter limit.
+;lease_renewal_increment = 1h
+```
+
+Example for `vault server -dev`:
+
+```ini
+[keystore.vault]
+url = http://127.0.0.1:8200 # HTTP should only be used for local testing
+auth_method = token
+token = s.sAZLyI0r7sFLMPq6MWtoOhAN # replace with your key
+```
+
+
+## Using the Vault expander
+
+When you've configured Vault you'll need to set the configuration or provisioning files you wish to
+use Vault. Vault configuration is an extension of configuration's [variable expansion]({{< relref "../administration/configuration.md#variable-expansion" >}}) and follows the
+`$__vault{<argument>}` syntax.
+
+The argument to Vault consists of three parts separated by a colon: the first part specifies which secrets engine
+should be used, the second part which secret should be accessed, and the third part which field of that secret
+should be used. For example, if you place a Key/Value secret for the Grafana admin user in _secret/grafana/admin_defaults_
+the syntax for accessing it's _password_ field would be `$__vault{kv:secret/grafana/admin_defaults:password}`.
+
+### Secrets engines
+
+Vault supports many secrets engines which represents different methods for storing or generating secrets when requested by an
+authorized user. Grafana supports a subset of these which are most likely to be relevant for a Grafana installation.
+
+#### Key/Value
+
+Grafana supports Vault's [K/V version 2](https://www.vaultproject.io/docs/secrets/kv/kv-v2) storage engine which
+is used to store and retrieve arbitrary secrets as `kv`. 
+
+```ini
+$__vault{kv:secret/grafana/smtp:username}
+```
+
+#### Databases
+
+Vault's [Databases secrets engines](https://www.vaultproject.io/docs/secrets/databases) is a family of
+secret engines which shares a similar syntax and grants the user dynamic access to a database.
+You can use this both for setting up Grafana's own database access and for provisioning data sources.
+
+```ini
+$__vault{database:database/creds/grafana:username}
+```
+
+### Examples
```

Comment:
Please add a brief intro to this section, explaining to the reader what the examples are of and how to use them.